src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -872,6 +889,11 @@ private void GeneratePropertySetter<T>(QualifiedType returnType, T decl, Class @
                     WriteLine("{0} = {1};", ctx.ReturnVarName, marshal.Context.Return);                    
                 }
 
+                if (arrayType != null && @class.IsValueType)
+                {
```

Comment:
Please do not use braces with one-liners. I personally prefer then but @tritao doesn't so we'd better do without.